src/pages/data_submission/RegistrationValidation.js

```diff
@@ -0,0 +1,106 @@
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import {dateValidator, emailValidator, urlValidator} from '../../components/forms/formValidation';
+import {get, isNil, set} from 'lodash';
+
+/**
+ * Generate a ValidateFunction from a dataset registration schema
+ * @param schema
+ * @returns ValidateFunction
+ */
+export const compileSchema = (schema) => {
+  const ajv = new Ajv({strict: false, allErrors: true});
+  addFormats(ajv);
+  ajv.addKeyword('label')
+    .addKeyword('prompt')
+    .addFormat('date', dateValidator.isValid)
+    .addFormat('uri', urlValidator.isValid)
+    .addFormat('email', emailValidator.isValid);
+
+  // Ajv doesn't like the `$schema` and `version` properties
+  // eslint-disable-next-line no-unused-vars
+  const {$schema, version, ...obj} = schema;
+  return ajv.compile(obj);
+};
+
+/**
+ * Validates given form data object according to the schema in a format that
+ * our internal form components can understand.
+ *
+ * @param {*} schema The dataset registration schema
+ * @param {*} formData Form data instance of registration schema
+ * @returns Form component compatible validation object
+ */
+export const validateForm = (schema, formData) => {
+  const validateFunction = compileSchema(schema);
+  const valid = validateFunction(formData);
+  const validation = errorsToValidation(validateFunction.errors);
+  return [valid, validation];
+};
+
+/**
+ * Construct a validation object of the form:
+ * {
+ *  <fieldName>: { failed: ['required'], valid: true|false},
+ *  <fieldName>: { failed: ['required'], valid: true|false},
+ * }
+ *
+ * @param errors Schema validation errors from Ajv
+ * @returns {{}}
+ */
+function errorsToValidation(errors) {
```

Comment:
This code, and below, is essentially lifted from the deleted `JsonSchemaUtils.js`. This way, there are no changes to how dependent UI components handle validation.